src/indicators/drought/wrsi.py

```diff
@@ -0,0 +1,309 @@
+"""Code to download and process WRSI data
+
+The WRSI data is made publicly available through
+the USGS website. The data is dekadal and each
+dekadal dataset is shared through zip files.
+https://edcftp.cr.usgs.gov/project/fews/dekadal/
+
+These include two separate zones for West Africa:
+
+- WA: the largest region, covering from the coast at
+    5 degrees to around 15 degrees latitude, ranging
+    from the western coast of Senegal to the western
+    border of Sudan. This is specifically for croplands:
+    https://earlywarning.usgs.gov/fews/product/56
+- W1: smaller, ranges from around 11 degrees latitude
+    to around 17, and from the eastern border of Chad
+    to Senegal. This is specifically for rangelands:
+    https://earlywarning.usgs.gov/fews/product/57
+
+Both are published using ESRI:102022. There is overlap between the
+two datasets where they don't agree, even though both are
+measuring the WRSI with respect to millet, due to their
+calculations for rangeland and cropland. Since most
+countries in the Sahel overlap with both datasets, both
+should be considered and we should get clarification from
+experts on the differences between the two, how they relate,
+and how best to use them.
+
+I have also just found the CHIRPS WRSI data on the USGS website.
+These are available at:
+
+https://edcftp.cr.usgs.gov/project/fews/africa/west/dekadal/wrsi-chirps-etos
+"""
+import datetime
+import logging
+import os
+from datetime import date
+from io import BytesIO
+from pathlib import Path
+from typing import List, Literal, Union, get_args
+from urllib.error import HTTPError
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import numpy as np
+import pandas as pd
+import rioxarray  # noqa: F401
+import xarray as xr
+from rasterio.crs import CRS
+
+logger = logging.getLogger(__name__)
+
+_BASE_URL = (
+    "https://edcftp.cr.usgs.gov/project/fews/"
+    "africa/west/dekadal/wrsi-chirps-etos/{region}/"
+)
+
+_URL_ZIP_FILENAME = "w{year:04}{dekad}{region}.zip"
+
+_BASE_FILENAME = "w{year:04}{dekad}{type}.tif"
+
+
+def _get_url_path(year, dekad, region):
+    if region == "cropland":
+        url_region, file_region = "west", "wa"
+    else:
+        url_region, file_region = "west1", "w1"
+
+    url_zip_filename = _URL_ZIP_FILENAME.format(
+        year=year, dekad=dekad, region=file_region
+    )
+
+    base_url = _BASE_URL.format(region=url_region)
+    return base_url + url_zip_filename
+
+
+def _download_wrsi(year: int, dekad: int, region: str, raw_dir: Path):
+    try:
+        url = _get_url_path(year=year, dekad=dekad, region=region)
+        resp = urlopen(url)
+    except HTTPError:
```

Comment:
So, I don't actually want to raise an error here, I should actually have as `logger.message`. In the previous instance of the data (pre-CHIRPS), there were certain missing years/dekads that weren't downloaded because they didn't exist, but isn't an actual error, just important to catch and log for the user if they might want to query why some data is missing. I put `logger.error` because I was avoiding changing my setting to show messages by default in console lol...But should probably be message so have changed! Hopefully isn't an issue anymore anyway with the CHIRPS data.